step-release-vis/src/app/services/environment_test.ts

```diff
@@ -15,7 +15,48 @@ describe('EnvironmentService', () => {
   });
 
   // TODO(naoai): write getPolygons test
-  it('#getPercentages should return 3 candidates with ~ equal percentages', () => {
+
+  it('#computeNextSnapshot insert new candidate', () => {
+    const inputCandInfo: CandidateInfo[] = [{name: '1', job_count: 100}, {name: '2', job_count: 100}];
+    const inputSet: TimestampUpperBoundSet = new TimestampUpperBoundSet();
+    inputSet.orderMap.set('1', 0);
+    inputSet.snapshot[0] = new PolygonUpperBoundYPosition('1', 100);
+
+    // @ts-ignore
+    const result: [TimestampUpperBoundSet, number] = service.computeNextSnapshot(inputCandInfo, inputSet);
+
+    expect(result[0].snapshot[0].position).toEqual(50);
+    expect(result[0].snapshot[1].position).toEqual(100);
+    expect(result[1]).toEqual(1);
+  });
+
+  it('#computeNextSnapshot candidateInfo[] empty', () => {
+    const emptyCandInfo: CandidateInfo[] = [];
+    const inputSet: TimestampUpperBoundSet = new TimestampUpperBoundSet();
+    inputSet.orderMap.set('1', 0);
+    inputSet.snapshot[0] = new PolygonUpperBoundYPosition('1', 100);
+
+    // @ts-ignore
+    const result: [TimestampUpperBoundSet, number] = service.computeNextSnapshot(emptyCandInfo, inputSet);
+
+    expect(result[0]).toBe(inputSet);
+  });
+
+  it('#computeNextSnapshot with empty set', () => {
```

Comment:
based on the name of the test I can not distinguish between this and the above one, lets be more descriptive.